src/components/rows/product.js

```diff
@@ -0,0 +1,77 @@
+      modal: false
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }       
+
+  toggle() {
+    this.setState({
+      modal: !this.state.modal
+    });
+  }
+
+  render() {
+    return (
+      <div>
+           <Image src={this.props.pix} alt="#" style={{width: '200px', height: '110px'}} onClick={this.toggle}  thumbnail/>
+              <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
+                  <ModalHeader toggle={this.toggle}>{this.props.title}</ModalHeader>
+                  <ModalBody style={{width: "100%", height: "30%"}}>
+                      <Container>
+                           <Row>
+                               <Col className="details">
+                                   <Image src={this.props.pix} style={{maxWidth: '100%', height: 'auto', display: 'block'}} thumbnail/>
+                                   <br /><br /><hr />
+                                   <div>
+                                       <h6 style={{marginTop: '17px'}}><b>Location:</b>{this.props.location}</h6>
+                                       
+                                       <h6 style={{marginTop: '17px'}}><b>Cost:</b> &nbsp; {this.props.tag}</h6>
+                                       
+                                       <h6 style={{marginTop: '17px'}}><b>Quantity available:</b>{this.props.quantity}</h6>
+                                       
+                                       <h6 style={{marginTop: '17px'}}><b>Phone no:</b> {this.props.phone}</h6>
+                                       
+                                       <h6 style={{marginTop: '17px'}}><b>Email:</b> {this.props.email}</h6>
+                                   </div>
+                               </Col>
+                           </Row>
+                      </Container>
+                 </ModalBody>
+                 <ModalFooter>
+                     <Button color="success" size="sm" onClick={this.toggle}>Buy Now</Button>{' '}
+                     <Button color="primary" size="sm" onClick={this.toggle}>Cancel</Button>
+                 </ModalFooter>
+              </Modal>
+             <div className="details">
+                   <p style={{marginTop: '5px'}}>
+                      <h6>
+                         {this.props.title}
+                      <br />
+                         <Badge color="secondary" style={{marginTop: '5px'}}>{this.props.tag}</Badge>
+                      </h6>
+                      <ButtonToolbar>
+                          <ButtonGroup>
+                              <Button className="button" color="success" size="sm" active>Buy Now</Button>{' '}
+                              <Button className="button" color="light" size="sm" onClick={this.toggle}>More Info</Button>{' '}
+                          </ButtonGroup>
+                      </ButtonToolbar>
+                   </p>
+              </div>
+      </div>
+    );
+  }
+}
+
+export default Products;
```

Comment:
So this was what I was talking about in the meeting. You are giving each item its own modal. Imagine now that you had 100, 1000, 10000 items on the screen and each one has a modal attached to it. The browser would have trouble attaching a modal to each item `div`.

Reply:
The solution here is to only have one modal and pass it in the appropriate information when you click on an item. Again one modal serving 0 to any number of items on our view
```
Step 1 - move the modal logic to its own component
2 - add modal the newly created modal component to the `anotherlist` component. it should be a sibling component to the `ListItems`
3 - writing a function that you pass into each modal component, that when clicked updates `anotherlist` internal state
4. the state update should tell the modal whether it should be open or not, and what contents it should display